consistency-checker/src/main/java/com/redhat/lightblue/migrator/consistency/MigrationJob.java

```diff
@@ -367,29 +370,77 @@ private int overwriteLightblue(List<JsonNode> documentsToOverwrite) {
     }
 
     protected boolean documentsConsistent(JsonNode sourceDocument, JsonNode destinationDocument) {
-        boolean consistent = true;
+        return doDocumentsConsistent(sourceDocument, destinationDocument, null);
+    }
 
-        Iterator<Entry<String, JsonNode>> nodeIterator = sourceDocument.fields();
+    //Recursive method
+    private boolean doDocumentsConsistent(final JsonNode sourceDocument, final JsonNode destinationDocument, final String path) {
+        List<String> excludes = getJobConfiguration().getComparisonExclusionPaths();
+        if(excludes != null && excludes.contains(path)) {
+            return true;
+        }
 
-        while (nodeIterator.hasNext()) {
-            Entry<String, JsonNode> sourceEntry = nodeIterator.next();
+        if(sourceDocument == null && destinationDocument == null){
+            return true;
+        }
+        else if(sourceDocument == null || destinationDocument == null){
+            return false;
+        }
 
-            List<String> excludes = getJobConfiguration().getComparisonExclusionPaths();
-            if(excludes == null || !excludes.contains(sourceEntry.getKey())) {
-                JsonNode sourceNode = sourceEntry.getValue();
-                JsonNode destinationNode = destinationDocument.findValue(sourceEntry.getKey());
+        if(JsonNodeType.ARRAY.equals(sourceDocument.getNodeType())){
+            if(!JsonNodeType.ARRAY.equals(destinationDocument.getNodeType())){
+                return false;
+            }
 
-                if((sourceNode == null || JsonNodeType.NULL.equals(sourceNode.getNodeType()))
-                        && (destinationNode == null || JsonNodeType.NULL.equals(destinationNode.getNodeType()))) {
-                    continue;
+            ArrayNode sourceArray = (ArrayNode) sourceDocument;
+            ArrayNode destinationArray = (ArrayNode) destinationDocument;
+
+            if(sourceArray.size() != destinationArray.size()){
+                return false;
+            }
+
+            //assumed positions in the array should be the same, else inconsistent.
+            for(int x = 0; x < sourceArray.size(); x++){
+                if(!doDocumentsConsistent(sourceArray.get(x), destinationArray.get(x), path)){
```

Comment:
Shouldn't `path` be updated to be path + "." + array field name? (or just array field name in case of top level field)